resources/asciidoctor/spec/elastic_compat_tree_processor_spec.rb

```diff
@@ -72,51 +74,82 @@
     expect(actual).to eq(expected.strip)
   end
 
-  [
-    %w[CONSOLE console],
-    %w[AUTOSENSE sense],
-    %w[KIBANA kibana],
-    %w[SENSE:path/to/snippet.sense sense],
-  ].each do |command, lang|
-    it "transforms legacy // #{command} commands into the #{lang} language" do
-      actual = convert <<~ASCIIDOC
-        == Example
-        [source,js]
-        ----
-        GET /
-        ----
-        pass:[// #{command}]
-      ASCIIDOC
-      expected = <<~DOCBOOK
-        <chapter id="_example">
-        <title>Example</title>
-        <programlisting language="#{lang}" linenumbering="unnumbered">GET /</programlisting>
-        </chapter>
-      DOCBOOK
-      expect(actual).to eq(expected.strip)
-    end
-  end
+  shared_examples 'snippet language' do |override, lang|
+    name = override ? " the #{override} lang override" : 'out a lang override'
+    context "for a snippet with#{name}" do
+      let(:snippet) do
+        snippet = <<~ASCIIDOC
+          [source,js]
+          ----
+          GET / <1>
+          ----
+        ASCIIDOC
+        snippet += "lang_override::[#{override}]" if override
+        snippet
+      end
+      let(:has_lang) do
+        /<programlisting language="#{lang}" linenumbering="unnumbered">/
+      end
+      shared_examples 'has the expected language' do
+        it "has the #{lang} language" do
+          expect(converted).to match(has_lang)
+        end
+      end
+      context 'when it is alone' do
+        let(:converted) do
+          convert <<~ASCIIDOC
+            == Example
+            #{snippet}
+          ASCIIDOC
+        end
+        include_examples 'has the expected language'
```

Comment:
This is very nuanced, but you could also use `it_behaves_like` here. `include_examples` is best suited for what you're doing from the topic level - creating specs using different values but when you have a nested context and want to emphasize a behavior, people tend to use `it_behaves_like`. They're effectively interchangeable and I think the difference really comes down to how you want the specs to print out. Sometimes it's also awkward to find a phrase that describes the behavior. For example, in this case, it might be strange to say `it_behaves_like 'a string converter'` - or whatever. So anyway, maybe `include_examples` is best here and you can ignore this whole comment = )

Reply:
Thanks for this! I looked at `it_behaves_like` for a bunch of things. I'll play with it some more and see what I can find. I found that it doesn't work well when you want to apply the same set of shared examples multiple times with different parameters. Or, rather, it doesn't seem to properly list the parameters when you run rspec. I'll try it some more though.

Reply:
I see what you're saying. Ultimately, they can accomplish very similar things in the specs, so it's up to you and how you want the the specs to run and print out.